src/algebra/algebra/restrict_scalars.lean

```diff
@@ -100,22 +99,32 @@ module structure over `R`.
 
 The preferred way of setting this up is `[module R M] [module S M] [is_scalar_tower R S M]`.
 -/
-instance : module R (restrict_scalars R S M) :=
+instance [module S M] : module R (restrict_scalars R S M) :=
 module.comp_hom M (algebra_map R S)
 
 /--
 This instance is only relevant when `restrict_scalars.module_orig` is available as an instance.
 -/
-instance : is_scalar_tower R S (restrict_scalars R S M) :=
+instance [module S M] : is_scalar_tower R S (restrict_scalars R S M) :=
 ⟨λ r S M, by { rw [algebra.smul_def, mul_smul], refl }⟩
 
 end
 
+instance restrict_scalars.op_module [module Sᵐᵒᵖ M] : module Rᵐᵒᵖ (restrict_scalars R S M) :=
+begin
+  letI : module Sᵐᵒᵖ (restrict_scalars R S M) := ‹module Sᵐᵒᵖ M›,
+  exact module.comp_hom M (algebra_map R S).op
+end
```

Comment:
Oh, I am surprised by this one!

If something happens to have both an S and an Smop action, why would restricting scalars for the S action also restrict the Smop action? In particular, a common application in my mind (from the theory of subfactors) is an inclusion of algebras A into B naturally giving us the A-B bimodule B. (Where we start with B as a B-B bimodule, and then restrict the left action.)

Reply:
I would have said that if you have commuting actions on the original module of any two rings, and you restrict one of them, you still have a commuting action with the other ring. (In fact, when I read the title and PR description this is what I was expecting to see.)

Reply:
The motivation for this is https://github.com/leanprover-community/mathlib/pull/10716, where I need both a left and right action in order for `restrict_scalars R A` to be considered an `S`-algebra.

> I would have said that if you have commuting actions on the original module of any two rings, and you restrict one of them, you still have a commuting action with the other ring.

Sure, but we can't even state the actions commute without first copying them across; and we can't copy across all of them, as:
* This would create diamonds if `M` already has an `S`-action
* This would entail copying across the `R`-action on `M` which you just removed.

Reply:
Discussion continued on [zulip](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/left.20vs.20right.20modules.20in.20tensor.20products/near/282033518).